src/main/java/Chelle.java

```diff
@@ -0,0 +1,90 @@
+                System.out.println("Chelle: ");
+                displayTasks(tasks, taskCount);
+            } else if (userInput.startsWith("mark")) {
+                markTask(userInput, tasks, taskCount);
+            } else if (userInput.startsWith("unmark")) {
+                unmarkTask(userInput, tasks, taskCount);
+            } else {
+                if (taskCount < tasks.length) {
+                    tasks[taskCount] = new Task(userInput);
+                    taskCount++;
+                    System.out.println("Chelle: added: " + userInput);
+                } else {
+                    System.out.println("Chelle: Task limit reached. Cannot add more tasks.");
+                }
+            }
+        }
+
+        scanner.close();
+    }
+
+    private static void displayTasks(Task[] tasks, int count) {
+        if (count == 0) {
+            System.out.println("No tasks to display.");
+        } else {
+            for (int i = 0; i < count; i++) {
+                Task task = tasks[i];
+                String status = task.isDone() ? "[X]" : "[ ]";
+                System.out.println((i + 1) + ". " + status + " " + task.getTaskName());
+            }
+        }
+    }
+
+    private static void markTask(String userInput, Task[] tasks, int taskCount) {
+        String[] parts = userInput.split(" ");
+        if (parts.length == 2) {
+            try {
+                int taskIndex = Integer.parseInt(parts[1]) - 1;
+                if (taskIndex >= 0 && taskIndex < taskCount) {
+                    tasks[taskIndex].markDone();
+                    System.out.println("Chelle: Nice! I've marked this task as done:\n        [X]  " +
+                            tasks[taskIndex].getTaskName());
+                } else {
+                    System.out.println("Chelle: Invalid task index.");
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Chelle: Invalid task index.");
+            }
+        } else {
+            System.out.println("Chelle: Invalid command format.");
+        }
+    }
+
+    private static void unmarkTask(String userInput, Task[] tasks, int taskCount) {
+        String[] parts = userInput.split(" ");
+        if (parts.length == 2) {
+            try {
+                int taskIndex = Integer.parseInt(parts[1]) - 1;
+                if (taskIndex >= 0 && taskIndex < taskCount) {
+                    tasks[taskIndex].markUndone();
+                    System.out.println("Chelle: OK, I've marked this task as not done yet:\n        [ ]  " +
+                            tasks[taskIndex].getTaskName());
+                } else {
+                    System.out.println("Chelle: Invalid task index.");
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Chelle: Invalid task index.");
+            }
+        } else {
+            System.out.println("Chelle: Invalid command format.");
+        }
```

Comment:
Can consider making this part more readable as it is arrowhead style code
![image](https://github.com/nus-cs2113-AY2324S2/ip/assets/24391419/72b67806-885b-4381-acd1-044d7a2bad82)

So try to settle all the bad cases first then perform the normal case 
if(badcase1){
// do badcase1 
}
else if(badcase2){
// do badcase2
}
else{
// do normal case
}